test/framework/src/main/java/org/elasticsearch/transport/nio/channel/TcpReadContext.java

```diff
@@ -75,7 +75,11 @@ public int read() throws IOException {
 
             try {
                 BytesReference messageWithoutHeader = message.slice(6, message.length() - 6);
-                handler.handleMessage(messageWithoutHeader, channel, channel.getProfile(), messageWithoutHeader.length());
+
+                // A message length of 6 bytes it is just a ping. Ignore for now.
```

Comment:
This happens in the `TcpFrameDecoder` [here](https://github.com/elastic/elasticsearch/blob/master/test/framework/src/main/java/org/elasticsearch/transport/nio/channel/TcpFrameDecoder.java#L62). If the header is invalid, an exception is thrown .

When the exception is caught, the channel goes through out normal exception handling (and disconnect) logic.